vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,862 @@
+
+```{r}
+confirmed_response <- function(datain) {
+
+  data_adrs <- datain %>%
+    group_by(USUBJID) %>%
+    mutate(AVALC.next = lead(AVALC),
+           AVAL.next = lead(AVAL),
+           ADT.next = lead(ADT)) %>%
+    mutate(AVALC.confirmed = case_when(
+      # better response
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        (is.na(NACTDT) | ADT.next <= NACTDT) &
+        AVAL.next >= AVAL ~ AVALC,
+      # worse response
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR", "SD") &
+        (is.na(NACTDT) | ADT.next <= NACTDT) &
+        AVAL.next < AVAL ~ AVALC.next,
+      # next assessment PD, NE, NA or after subsequent therapy
+      AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") &
+        (AVALC.next == "PD" | AVALC.next == "NE" | is.na(AVALC.next) |
+           !is.na(NACTDT) & ADT.next > NACTDT) ~ "SD",
+      # no need to confirm SD and NE
+      AVALC %in% c("SD", "NE") ~ AVALC,
+      # confirmed progression
+      AVALC == "PD" &
+        (PDIFL == "Y" | DTHPDFL == "Y" | PDOFL == "Y" & AVALC.next == "PD") ~
+        AVALC,
+      # unconfirmed progression
+      AVALC == "PD" & is.na(DTHPDFL) & PDOFL == "Y" &
+        (AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR", "SD", "NE") |
+           is.na(AVALC.next)) ~ "NE")) %>%
+      mutate(AVAL.confirmed = aval_resp_conf(AVALC.confirmed)) %>%
+      # best Confirmed Response so far
+      mutate(AVAL.confirmed = cummax(AVAL.confirmed)) %>%
+      ungroup(USUBJID)
+
+  # char mapping to go back to AVALC values
+  avalc_resp_conf <- function(arg) {
+    case_match(
+      arg,
+      8 ~ "PD",
+      7 ~ "sCR",
+      6 ~ "CR",
+      5 ~ "VGPR",
+      4 ~ "PR",
+      3 ~ "MR",
+      2 ~ "SD",
+      1 ~ "NE",
+      NA_real_ ~ NA
+    )
+  }
+
+  data_adrs <- data_adrs %>%
+      mutate(AVALC.confirmed = avalc_resp_conf(AVAL.confirmed)) %>%
+      # if no Confirmed Response, assign SD
+      mutate(AVALC.confirmed = case_when(
+        AVALC.confirmed == "NE" ~ "SD",
+        TRUE ~ AVALC.confirmed
+      ))%>%
+      select(-AVAL, -AVALC, -AVAL.next, -AVALC.next, -ADT.next, 
+             -AVAL.confirmed) %>%
+      rename(AVALC = AVALC.confirmed) %>%
+      mutate(AVAL = aval_resp_imwg(AVALC)) %>%
+      mutate(
+        PARAMCD = "COVR",
+        PARAM = "Confirmed Response at Time Point by Investigator",
+        PARCAT1 = "Investigator",
+        PARCAT2 = "IMWG"
+      )
+}
+adrs_imwg <- confirmed_response(adrs) 
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs_imwg,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVALC)
+)
+```
+
+## Analysis flag derivation {#anlfl}
+
+### Flag One Assessment at Each Analysis Visit (`ANL01FL`)  {#anl01fl}
+
+To confirm responses, we took into account all assessments - 
```

Comment:
can we clarify and say " to get confirm responses on each visit.." in the start of the sentence

Reply:
ok